### DISCUSSION TITLE
Multi-warehouse API

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2831] [Feature Requests]
### What
Allow merchants to manage inventory across locations and enable new customer experiences

### Motivation
Merchants with multiple stock-keeping locations want to manage their inventory at each location and make it available for different sales channels and fulfillment methods. Currently, Medusa only holds a single inventory count. This forces merchants to either make heavy customizations to Medusa or use 3rd parties if they need to make fulfillments across locations or restrict inventory across locations.

**Comment by abhiraj-zangoh** [comment]
@olivermrbl Is it possible to allow customers to choose a specific location and display products available only in that location? Each store or warehouse would have its own separate inventory and pricing, ensuring accurate product availability based on the selected location.
